ddp-workspace/projects/ddp-dsm-ui/src/app/services/dsm.service.ts

```diff
@@ -234,8 +234,11 @@ export class DSMService {
     } else if (viewFilterCopy != null) {
       map.push({name: 'filters', value: JSON.stringify(viewFilterCopy.filters)});
     }
-    const filters = JSON.parse(jsonPatch);
-    const body = {columnNames: columns, ...filters};
+    let body = {columnNames: columns};
+    if (jsonPatch) {
+      const filters = JSON.parse(jsonPatch);
+      body = {columnNames: columns, ...filters};
+    }
```

Comment:
**_#optionaly_**
```
 let body = {columnNames: columns};
    if (jsonPatch) {
      const filters = JSON.parse(jsonPatch);
      body = {columnNames: columns, ...filters};
    }
```
can be simplified like:
```
const filters = jsonPatch ? JSON.parse(jsonPatch) : undefined;
const body = {columnNames: columns, ...filters};
```

Reply:
thank you! I appreciate the improvement